source/d3d8to9.cpp

```diff
@@ -18,7 +18,10 @@ std::ofstream LOG;
 extern "C" Direct3D8 *WINAPI Direct3DCreate8(UINT SDKVersion)
 {
 #ifndef D3D8TO9NOLOG
-	LOG.open("d3d8.log", std::ios::trunc);
+	if (!LOG.is_open())
+	{
+		LOG.open("d3d8.log", std::ios::trunc);
+	}
 
 	if (!LOG.is_open())
 	{
```

Comment:
Changing the screen resolution does not require recreation of the entire D3D8 COM interface. I'd be surprised if any game calls this function at any other time other than startup (it may call it multiple times there, but it's unlikely to happen again once it's running).